mypy_django_plugin/main.py

```diff
@@ -42,7 +42,8 @@ def transform_model_class(ctx: ClassDefContext, django_context: DjangoContext) -
     sym = ctx.api.lookup_fully_qualified_or_none(fullnames.MODEL_CLASS_FULLNAME)
 
     if sym is not None and isinstance(sym.node, TypeInfo):
-        helpers.get_django_metadata(sym.node)["model_bases"][ctx.cls.fullname] = 1
+        bases = helpers.get_django_metadata(sym.node).setdefault("model_bases", {})
+        bases[ctx.cls.fullname] = 1
```

Comment:
I'm not sure why there's a number(`1`) set as value for these. All I've encountered is they existence of the fullname as key and value is ignored.

Not saying we should change the type, but what do you think about hiding the `= 1` behind some method interface?

e.g.

```python
class DjangoTypeMetadata(...):
    ...

    def add(self, fullname: str, /) -> None:
        self[fullname] = 1
```

Just thinking that might allow later refactoring where e.g. the `1` could be removed.

Reply:
Can't add methods to a TypedDict.

These should be converted to `set` instead of `dict` anyway, which already has an `add` method, so the additional method would end up unnecessary.

Reply:
I could refactor these to `set`, but without understanding of this code, I feel that increases risk to introduce new bugs.

Does this metadata end up serialized into `.mypy_cache`? Seems like yes. Then that would cause backwards incompatibility unless we somehow also flush the cache. Then how do I do that?

I'd rather not deal with this scope creep, I'm just trying to fix an obvious bug here.

Reply:
> Can't add methods to a TypedDict.
> ...

> I could refactor these to `set`, but without understanding of this code, I feel that increases risk to introduce new bugs.
> 
> Does this metadata end up serialized into `.mypy_cache`? Seems like yes. Then that would cause backwards incompatibility unless we somehow also flush the cache. Then how do I do that?
> 
> I'd rather not deal with this scope creep, I'm just trying to fix an obvious bug here.

No need to refactor to a `set`. I still meant to keep the dict type with an int. I only meant for later changes, and perhaps get rid of seeing/setting the `1` explicitly now.

Just an additional comment in regards to cache, I would suspect you'd have to explicitly tell people to flush cache.
Though I'm not 100% sure, but I have previously gotten the impression that mypy taints the whole cache when a plugin version has changed. But I have nothing to back that up with.